### PR TITLE
Implement caliptra_builder::build_and_sign_image.

### PR DESCRIPTION
**Pull request by korran**
This is used for building fmc/app-fw bundles for use in integration
tests.

### CODE REVIEW

File: rom/dev/tests/smoke_test.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// Licensed under the Apache-2.0 license
+
+use caliptra_builder::{ImageOptions, APP_WITH_UART, FMC_WITH_UART, ROM_WITH_UART};
+use caliptra_hw_model::{HwModel, InitParams};
+
+#[test]
+fn smoke_test() {
```
Comment (rusty1968) [inline rom/dev/tests/smoke_test.rs:7]: Where should the integration test bench with the runtime and fmc crates reside?

File: rom/dev/tests/smoke_test.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        &FMC_WITH_UART,
+        &APP_WITH_UART,
+        ImageOptions::default(),
+    )
+    .unwrap();
+    let mut hw = caliptra_hw_model::create(InitParams {
+        rom: &rom,
+        ..Default::default()
+    })
+    .unwrap();
```
Comment (rusty1968) [inline rom/dev/tests/smoke_test.rs:19]: Should we step until flow_status.ready_for_fw  is set?

Comment (korran) [reply rom/dev/tests/smoke_test.rs:19]: I want upload_firmware() to do that.

File: rom/dev/tests/smoke_test.rs (hunk 1 above)
Comment (rusty1968) [reply rom/dev/tests/smoke_test.rs:7]: Agree. This is a very useful outcome.

Comment (korran) [reply rom/dev/tests/smoke_test.rs]: As there's still a lot of unimplemented stuff here, I've decided to remove this commit from the PR. Will follow up in a future PR.

File: rom/dev/tests/smoke_test.rs (hunk 2 above)
Comment (korran) [reply rom/dev/tests/smoke_test.rs]: Removed from this PR.

**Review by rusty1968 — DISMISSED** [review]

**Review by rusty1968 — APPROVED** [review]

**Pull request merged**